indexer/string_file_values.hpp

```diff
@@ -18,138 +23,236 @@
 /// A wrapper around feature index.
 struct FeatureIndexValue
 {
-  FeatureIndexValue() : m_value(0) {}
+  FeatureIndexValue() : m_featureId(0) {}
+
+  FeatureIndexValue(uint64_t featureId) : m_featureId(featureId) {}
 
+  // The serialization and deserialization is needed for StringsFile.
+  // Use ValueList for group serialization in CBVs.
   template <typename TWriter>
-  void Write(TWriter & writer) const
+  void Serialize(TWriter & writer) const
   {
-    WriteToSink(writer, m_value);
+    WriteToSink(writer, m_featureId);
   }
 
   template <typename TReader>
-  void Read(TReader & reader)
+  void Deserialize(TReader & reader)
+  {
+    ReaderSource<TReader> src(reader);
+    DeserializeFromSource(src);
+  }
+
+  template <typename TSource>
+  void DeserializeFromSource(TSource & src)
   {
-    m_value = ReadPrimitiveFromSource<uint32_t>(reader);
+    m_featureId = ReadPrimitiveFromSource<uint64_t>(src);
   }
 
-  inline void const * data() const { return &m_value; }
+  inline void const * data() const { return &m_featureId; }
 
-  inline size_t size() const { return sizeof(m_value); }
+  inline size_t size() const { return sizeof(m_featureId); }
 
-  bool operator<(FeatureIndexValue const & value) const { return m_value < value.m_value; }
+  bool operator<(FeatureIndexValue const & o) const { return m_featureId < o.m_featureId; }
 
-  bool operator==(FeatureIndexValue const & value) const { return m_value == value.m_value; }
+  bool operator==(FeatureIndexValue const & o) const { return m_featureId == o.m_featureId; }
 
-  void swap(FeatureIndexValue & value) { ::swap(m_value, value.m_value); }
+  void Swap(FeatureIndexValue & o) { ::swap(m_featureId, o.m_featureId); }
```

Comment:
Please, keep old unix_hacker_style for swap(), or change style to CamelCase for data() and size().